server/test/buff_server/packages_test.exs

```diff
@@ -1,90 +1,103 @@
 defmodule BuffServer.PackagesTest do
+
+  @s3_bucket_name Application.get_env(:buff_server, :s3_bucket_name)
   use BuffServer.DataCase, async: true
+  import Mox
 
   alias BuffServer.Packages
 
   describe "packages" do
-    alias BuffServer.Packages.Package
 
     @valid_attrs %{
       description: "some description",
       homepage: "some homepage",
       keywords: [],
-      name: "some name",
-      repository_url: "http://aaa.com"
-    }
-    @update_attrs %{
-      description: "some updated description",
-      homepage: "some updated homepage",
-      keywords: [],
-      name: "some updated name",
-      repository_url: "http://bbb.com"
-    }
-    @invalid_attrs %{
-      description: nil,
-      homepage: nil,
-      keywords: nil,
-      name: nil,
-      repository_url: nil
+      name: "some-name",
+      repository_url: "http://aaa.com",
+      s3_bucket_name: @s3_bucket_name,
+      s3_bucket_path: "/some-name/artifact"
     }
+    # @update_attrs %{
+    #   description: "some updated description",
+    #   homepage: "some updated homepage",
+    #   keywords: [],
+    #   name: "some updated name",
+    #   repository_url: "http://bbb.com"
+    # }
+    # @invalid_attrs %{
+    #   description: nil,
+    #   homepage: nil,
+    #   keywords: nil,
+    #   name: nil,
+    #   repository_url: nil
+    # }
 
-    def package_fixture(attrs \\ %{}) do
+    def artifact_fixture() do
+      {:ok, artifact_binary} = File.read("test/buff_server/fixtures/dummy_artifact.gz")
+      artifact_binary
+    end
+
+    def package_fixture(attrs, artifact_binary) do
```

Comment:
I would avoid having both attrs and another argument.. pattern matching looks more explicit